gammapy/utils/scripts.py

```diff
@@ -133,23 +140,70 @@ def _configure_root_logger(level='info', format=None):
 
 
 def read_yaml(filename, logger=None):
-    """Read config from YAML file."""
+    """
+    Read config from YAML file 
+    """
     import yaml
     if logger is not None:
         logger.info('Reading {}'.format(filename))
     with open(filename) as fh:
         config = yaml.safe_load(fh)
+
     return config
 
 
 def write_yaml(config, filename, logger=None):
-    """Write YAML config file
+    """
+    Write YAML config file
 
     This function can be used by scripts that alter the users config file.
+
+    Parameters
+    ----------
+    config : dict
+        config to write
+    filename : str, `~gammapy.exter.pathlib.Path`
+        file to write
     """
     import yaml
-    filename = filename + '.yaml'
+    filename = make_path(filename)
     if logger is not None:
         logger.info('Writing {}'.format(filename))
-    with open(filename, 'w') as outfile:
+    with open(str(filename), 'w') as outfile:
         outfile.write(yaml.dump(config, default_flow_style=False))
+
+
+def make_path(path):
+    """
+    Expand environment varibles on `~pathlib.Path` construction
+
+    Parameters
+    ----------
+    path : str, `~gammapy.extern.pathlib.Path`
+        path to expand
+    """
+    return Path(expandvars(str(path)))
+
+
+def recursive_update_dict(old, new):
+    """Recursively update a dict with new values
+
+    The built-in update function cannot be used for hierarchical dicts
+    where only a subset of keys shall be updated.
+    see: http://stackoverflow.com/questions/3232943/update-value-of-a-nested-dictionary-of-varying-depth
```

Comment:
Maybe take this code and link to it directly?
http://stackoverflow.com/a/3233356